gateway/api_loader.go

```diff
@@ -728,9 +726,23 @@ func (gw *Gateway) loadHTTPService(spec *APISpec, apisByListen map[string]int, g
 
 	subrouter := router.PathPrefix(spec.Proxy.ListenPath).Subrouter()
 
-	chainObj := gw.processSpec(spec, apisByListen, gs, subrouter, logrus.NewEntry(log))
+	var chainObj *ChainObject
+
+	gw.apisMu.RLock()
+	if curSpec, ok := gw.apisByID[spec.APIID]; ok && curSpec.Checksum == spec.Checksum {
```

Comment:
isn't there already a safe access function to this var?

Reply:
yes, changed to using that